FindmallocMC.cmake

```diff
@@ -96,8 +93,14 @@ if(mallocMC_ROOT_DIR)
     string(REGEX MATCH "([0-9]+)" mallocMC_VERSION_PATCH
                                 ${mallocMC_VERSION_PATCH_HPP})
```

Comment:
just because I found it here (sry): pls unset the tmp vars `mallocMC_VERSION_MAJOR_HPP`, `mallocMC_VERSION_MINOR_HPP` and `mallocMC_VERSION_PATCH_HPP` (I got this feedback while porting a similar patch upstream for `FindHDF5`)